pkg/networkmanager/interfaces.js

```diff
@@ -766,24 +790,15 @@ function NetworkManagerModel() {
     var connections_by_uuid = { };
 
     function set_settings(obj, settings) {
-        var cs = connection_settings(obj);
-        if (cs.uuid)
-            delete connections_by_uuid[cs.uuid];
+        if (obj.Settings && obj.Settings.connection && obj.Settings.connection.uuid)
+            delete connections_by_uuid[obj.Settings.connection.uuid];
```

Comment:
Should other consumers of `connection_settings()` follow this pattern as well (i. e. "look before you leap")? If so, with this expectation the `connection_settings()` function should be eliminiated. Or does its fallback of returning `{}` still make sense for some cases where the absence of settings is not a race, but we are dealing with unconfigured  connections?